src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -314,7 +319,7 @@ impl ThinPool {
                     // A more sophisticated approach might be in order.
                     match self.extend_thinpool(dm, usage.total_data, bd_mgr) {
                         #![allow(single_match)]
-                        Ok(_) => {}
+                        Ok(_) => should_save = true,
```

Comment:
If you don't do the same thing for the ```Ok``` as for the ```Err```, you can remove the ```single_match``` clippy annotation.

Reply:
great! ok.